bertopic/representation/_openai.py

```diff
@@ -217,11 +218,21 @@ def extract_topics(self,
                 else:
                     response = self.client.chat.completions.create(**kwargs)
 
-                # Check whether content was actually generated
-                # Adresses #1570 for potential issues with OpenAI's content filter
-                if hasattr(response.choices[0].message, "content"):
-                    label = response.choices[0].message.content.strip().replace("topic: ", "")
+                choice = response.choices[0]
+
+                if choice.finish_reason == "stop":
+                    label = choice.message.content.strip().replace("topic: ", "")
+                elif choice.finish_reason == "length":
+                    logger.warn(f"Extracing Topics - Length limit reached for doc_ids ({repr_doc_ids})")
+                    if hasattr(response.choices[0].message, "content"):
+                        label = choice.message.content.strip().replace("topic: ", "")
+                    else:
+                        label = "Incomple output due to token limit being reached"
+                elif choice.finish_reason == "content_filter":
+                    logger.warn(f"Extracing Topics - Content filtered for doc_ids ({repr_doc_ids})")
+                    label = "Output content filtered by OpenAI"
                 else:
+                    logger.warn(f"Extracing Topics - No label due to finish_reason {choice.finish_reason} for doc_ids ({repr_doc_ids})")
```

Comment:
I am not in love with exposing the `finish_reason` concept in the error message but I wonder with folks will have to learn about it when debugging the issue anyway. Wyt?

Reply:
I also think I need to index into `repr_doc_ids` based on the index. Kudos to @sdehm for pointing that out.

Reply:
> I am not in love with exposing the finish_reason concept in the error message but I wonder whether folks will have to learn about it when debugging the issue anyway. Wyt?

If that reason didn't trigger any of the other if statements, then to me it makes sense to log it since we didn't expect it and there is no reason for the user to know which one it is. 

Also, perhaps make the warning a bit more explicit. For instance, "OpenAI Topic Representation - Couldn't create a label due to {choice.finish_reason} for the following document IDs"

I think being explicit here might improve any potential debugging.